### PR TITLE
Feature/brain key support

### PR DESCRIPTION
**Pull request by lukas1**

### CODE REVIEW

File: Sources/DCoreKit/Crypto/BrainKey.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+     - Returns: `BrainKey` consisting of random words.
+     */
+    public static func generate(
+        usingSeedDictionary seedDictionary: [String],
+        count: Int = DCore.Constant.brainKeyWordCount) -> BrainKey {
+        return generate(
+            withEntropy: CryptoUtils.secureRandom(), seedDictionary: seedDictionary, count: count)
+    }
+
+    static func generate(withEntropy entropy: Data, seedDictionary: [String], count: Int) -> BrainKey {
```
Comment (fl3xman) [inline Sources/DCoreKit/Crypto/BrainKey.swift:32]: Should not be private?

File: Sources/DCoreKit/Crypto/BrainKey.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        var words = [String]()
+        for iteration in stride(from: 0, to: count * 2, by: 2) {
+            // entropy buffer has 256 bits / 16 bits per word == 16 words
+            let num = (UInt64(entropy[iteration]) << 8) + UInt64(entropy[iteration + 1])
+            
+            // convert into a int between 0 and 1 (inclusive)
+            let multiplier = Double(num) / pow(2.0, 16.0)
+            let index = abs(round(Double(seedDictionary.count) * multiplier))
```
Comment (fl3xman) [inline Sources/DCoreKit/Crypto/BrainKey.swift]: Abs not required

File: Sources/DCoreKit/Crypto/BrainKey.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    private static func loadDefaultSeedDictionary() -> [String] {
+        guard let dictionaryPath = Bundle(for: DCore.Api.self).path(forResource: "SeedDictionary", ofType: "txt") else {
+            preconditionFailure("Failed to load brain key seed dictionary")
+        }
+
+        let seedDictionary: [String]
+        do {
+            let fileContent = try String(contentsOfFile: dictionaryPath, encoding: String.Encoding.utf8)
```
Comment (fl3xman) [inline Sources/DCoreKit/Crypto/BrainKey.swift]: shorthand encoding: .utf8

File: Sources/DCoreKit/Crypto/BrainKey.swift (hunk 1 above)
Comment (lukas1) [reply Sources/DCoreKit/Crypto/BrainKey.swift:32]: no, because I want it to be testable

**Review by fl3xman — APPROVED** [review]

**Pull request merged**